dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LDAPMemberAttributeScopeTest.java

```diff
@@ -0,0 +1,577 @@
+        ldapRegistry.setLdapCache(new LdapCache(new AttributesCache(false, null, null, null), new SearchResultsCache(false, null, null, null)));
+        ldapRegistry.getLdapEntityTypes().add(new LdapEntityType("Group", null, new String[] { "groupofnames", "groupofurls" }, null));
+        ldapRegistry.getLdapEntityTypes().add(new LdapEntityType("PersonAccount", null, new String[] { "inetorgperson" }, null));
+
+        GroupProperties groupProperties = new GroupProperties();
+        ldapRegistry.setGroupProperties(groupProperties);
+        groupProperties.setDynamicMemberAttribute(new DynamicMemberAttribute("memberurl", "groupofurls"));
+
+        if (memberAttrs != null && memberAttrs.length > 0) {
+            ldapRegistry.setRecursiveSearch(true); // TODO This should happen automatically.
+            for (MemberAttribute memberAttr : memberAttrs) {
+                groupProperties.getMemberAttributes().add(memberAttr);
+            }
+        }
+
+        if (membershipAttr != null) {
+            ldapRegistry.setRecursiveSearch(true); // TODO This should happen automatically.
+            groupProperties.setMembershipAttribute(membershipAttr);
+        }
+
+        return ldapRegistry;
+    }
+
+    /**
+     * Test a memberAttribute whose scope is set to "direct". Direct scoped member attributes
+     * only return their direct members, thus this will require nested calls to derive
+     * group membership.
+     *
+     * @throws Exception if the test failed for some unforeseen reason.
+     */
+    @Test
+    public void memberAttribute_scope_direct() throws Exception {
+
+        ServerConfiguration clone = baseConfiguration.clone();
+        MemberAttribute[] memberAttrs = new MemberAttribute[1];
+        memberAttrs[0] = new MemberAttribute(null, "member", "groupofnames", LdapConstants.LDAP_DIRECT_GROUP_MEMBERSHIP_STRING);
+        createLdapRegistry(clone, MEMBER_BASE_DN, memberAttrs, null);
+        LDAPFatUtils.updateConfigDynamically(libertyServer, clone);
+
+        List<String> results = urServlet.getGroupsForUser(MEMBER_LEVEL1_USER_DN);
+        assertMembershipResults("With direct scope, the direct member should only be a member of the root group.", results, MEMBER_LEVEL1_GROUP_DN);
+
+        results = urServlet.getGroupsForUser(MEMBER_LEVEL2_USER_DN);
+        assertMembershipResults("With direct scope, the nested member should be a member of the root and nested group.", results, MEMBER_LEVEL1_GROUP_DN, MEMBER_LEVEL2_GROUP_DN);
+
+        results = urServlet.getGroupsForUser(MEMBER_DYNAMIC_USER_DN);
+        assertMembershipResults("With direct scope, the dynamic member should be a member of the root and dynamic group.", results, MEMBER_LEVEL1_GROUP_DN,
+                                MEMBER_DYNAMIC_GROUP_DN);
+
+        results = urServlet.getUsersForGroup(MEMBER_LEVEL1_GROUP_DN, 0).getList();
```

Comment:
The LdapAdapter.getMembersByMember() method does not consider nested dynamic groups.